cl_kernel/kernel_geo_map.cl

```diff
@@ -79,6 +80,11 @@ kernel_geo_map (
     bool out_of_bound[8];
     float2 input_pos[8];
     // map to [-0.5, 0.5)
+    if (g_x * PIXEL_RES_STEP_X < out_size.x / 2.0f)
+        table_scale_size *= left_scale_factor;
+    else if (g_x * PIXEL_RES_STEP_X > out_size.x / 2.0f)
+        table_scale_size *= right_scale_factor;
```

Comment:
consider select instead of if/else branch.
table_scale_size *= (g_x * PIXEL_RES_STEP_X < out_size.x / 2.0f) ?  left_scale_factor : right_scale_factor;